### PR TITLE
Create PipelineDataSourceFactoryTest.java

### PR DESCRIPTION
**Pull request by sunkai-cai**
complete #15693.

Changes proposed in this pull request:
- add test for PipelineDataSourceFactory 
- PipelineDataSourceCreatorFactory has test

### CODE REVIEW

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import org.junit.Test;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+public class PipelineDataSourceFactoryTest {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:31]: Please add final is class if not design for extend

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    @Test
+    public void testNewInstance() {
+        Map<String, Object> yamlDataSourceConfig = new HashMap<>();
+        yamlDataSourceConfig.put("url", "jdbc:mysql://localhost:3306/database");
+        yamlDataSourceConfig.put("username", "username");
+        yamlDataSourceConfig.put("password", "password");
+        StandardPipelineDataSourceConfiguration standardPipelineDataSourceConfiguration = new StandardPipelineDataSourceConfiguration(yamlDataSourceConfig);
+        PipelineDataSourceWrapper pipelineDataSourceWrapper = PipelineDataSourceFactory.newInstance(standardPipelineDataSourceConfiguration);
+        assertNotNull(pipelineDataSourceWrapper);
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:41]: Please remove unnecessary assertion

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertNotNull;
+
+public class PipelineDataSourceFactoryTest {
+
+    @Test
+    public void testNewInstance() {
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:34]: Please use `assert` as prefix for test method

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        Map<String, Object> yamlDataSourceConfig = new HashMap<>();
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:35]: Please add initial size of HashMap

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java (hunk 4 above)
Comment (sunkai-cai) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:35]: ok

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java (hunk 3 above)
Comment (sunkai-cai) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:34]: oh, I forgot.

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java (hunk 2 above)
Comment (sunkai-cai) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:41]: ok

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java (hunk 1 above)
Comment (sunkai-cai) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:31]: ok

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.Map;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class PipelineDataSourceFactoryTest {
+
+    @Test
+    public void assertNewInstance() {
+        Map<String, Object> yamlDataSourceConfig = new HashMap<>(3);
```
Comment (terrymanu) [inline shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:34]: new HashMap<>(3, 1) is better.
The default factory is 0.75.

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java (hunk 5 above)
Comment (sunkai-cai) [reply shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/test/java/org/apache/shardingsphere/data/pipeline/core/datasource/PipelineDataSourceFactoryTest.java:34]: thanks the tip

**Pull request merged**